### PR TITLE
Csm transition v3.7.0

### PR DESCRIPTION
**Pull request by troublemaker52025**
Re: #

### Changelog

*

-------

### Continuous Integration

This pull request:

* [x] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [x] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by troublemaker52025** [comment]
@cocos-robot run test cases

File: cocos/3d/lights/directional-light-component.ts [hunk 1]
```diff
@@ -446,6 +451,45 @@ export class DirectionalLight extends Light {
         }
     }
 
+    /**
+     * @en Enabled shadow advanced options
```
Comment (minggo) [inline cocos/3d/lights/directional-light-component.ts:465]: What's `advanced options` mean? What advanced features?

File: cocos/3d/lights/directional-light-component.ts [hunk 2]
```diff
@@ -446,6 +451,45 @@ export class DirectionalLight extends Light {
...
+    set csmAdvancedOptions (val) {
+        this._csmAdvancedOptions = val;
+    }
+
+    /**
+     * @en Enabled csm layers transition
+     * @zh 是否启用级联阴影层级过渡？
+     */
+    @visible(function (this: DirectionalLight) {
+        return (legacyCC.director.root as Root).pipeline.pipelineSceneData.shadows.enabled
```
Comment (minggo) [inline cocos/3d/lights/directional-light-component.ts]: Now core only export `cclagacy`.

File: cocos/3d/lights/directional-light-component.ts (hunk 1 above)
Comment (troublemaker52025) [reply cocos/3d/lights/directional-light-component.ts:465]: This option will improve quality and reduce performance. In the future we will simplify the shadow-related settings, collapsing some unnecessarily complex options into the advanced options

File: cocos/3d/lights/directional-light-component.ts (hunk 2 above)
Comment (troublemaker52025) [reply cocos/3d/lights/directional-light-component.ts]: sure

File: cocos/3d/lights/directional-light-component.ts (hunk 1 above)
Comment (minggo) [reply cocos/3d/lights/directional-light-component.ts:465]: Ok. What i mean is that, should add these information in docs.

Comment (troublemaker52025) [reply cocos/3d/lights/directional-light-component.ts:465]: sure

**Comment by troublemaker52025** [comment]
* https://github.com/cocos/cocos-engine/pull/12974

**Review by jk20012001 — APPROVED** [review]

File: native/cocos/renderer/pipeline/shadow/ShadowFlow.cpp [hunk 3]
```diff
@@ -98,6 +98,9 @@ void ShadowFlow::activate(RenderPipeline *pipeline) {
     // 0: CC_DIR_LIGHT_SHADOW_PLANAR, 1: CC_DIR_LIGHT_SHADOW_UNIFORM, 2: CC_DIR_LIGHT_SHADOW_CASCADED, 3: CC_DIR_LIGHT_SHADOW_VARIANCE
     pipeline->setValue("CC_DIR_LIGHT_SHADOW_TYPE ", 0);
 
+    // 0: CC_CASCADED_LAYERS_TRANSITION_OFF, 1: CC_CASCADED_LAYERS_TRANSITION_ON
+    pipeline->setValue("CC_CASCADED_LAYERS_TRANSITION ", 0);
```
Comment (dumganhar) [inline native/cocos/renderer/pipeline/shadow/ShadowFlow.cpp:102]: Why adding a `space` after CC_CASCADED_LAYERS_TRANSITION?

**Pull request merged**